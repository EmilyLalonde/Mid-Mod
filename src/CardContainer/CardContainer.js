import React from 'react';
import Card from '../Card/Card.js'
import '../CardContainer/CardContainer.css'

const CardContainer = ({data}) => {
  const allPurchases = data.map(purchase => {
    return (
      <Card 
      id = {purchase.id}
      img = {purchase.img}
      name = {purchase.name}
      description = {purchase.description}
      price = {purchase.price }
      />
    )
  })
  return (
    <div>
      {allPurchases}
    </div>
  )
  
}

export default CardContainer