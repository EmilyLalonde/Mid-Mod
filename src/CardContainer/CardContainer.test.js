import React from 'react'
import { shallow } from 'enzyme'
import CardContainer from './CardContainer.js'

describe('CardContainer', () => {
  
  it('should match the snapshot', () => {
    const mockPurchases = [
      {
      description: 'The latest phone to come out by Apple. It has a faster processor and improved camera to take HDR images.',
      id: 1, 
      img: 'https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/iphone-xs-max-gold-select-2018?wid=940&hei=1112&fmt=png-alpha&qlt=80&.v=1550795409154',
      key: 1,
      name: 'iPhone Xs',
      price: 900
      }
    ]
    const wrapper = shallow(<CardContainer data={mockPurchases}/>)

    expect(wrapper).toMatchSnapshot();
  });
})
