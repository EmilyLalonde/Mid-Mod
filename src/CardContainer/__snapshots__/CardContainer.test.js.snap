// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
<div>
  <Card
    description="The latest phone to come out by Apple. It has a faster processor and improved camera to take HDR images."
    id={1}
    img="https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/iphone-xs-max-gold-select-2018?wid=940&hei=1112&fmt=png-alpha&qlt=80&.v=1550795409154"
    key="1"
    name="iPhone Xs"
    price={900}
  />
</div>
`;
